src/chessboard/components/Piece.tsx

```diff
@@ -160,6 +160,7 @@ export function Piece({
           viewBox={"1 1 43 43"}
           width={boardWidth / 8}
           height={boardWidth / 8}
+          style={{ display: "block" }}
```

Comment:
The issue is that the SVG is overflowing the DIV, mainly due to SVG elements being display: inline; by default. This can easily be fixed by adding a display property to the SVG's styles, such as display: block;